app/controllers/course/material/folders_controller.rb

```diff
@@ -59,11 +62,14 @@ def upload_materials
           redirect_to course_material_folder_path(current_course, @folder),
                       success: t('.success', name: @folder.name)
         end
-        format.json
+        format.json do
+          show
+          render 'show', status: :ok
+        end
       else
         format.html { upload_materials_failure }
         format.json do
-          render json: { message: @folder.errors.full_messages.to_sentence },
+          render json: { message: @folder.errors },
```

Comment:
`@folder.errors.full_messages.to_sentence` is correct. We only return `.errors` for form related action (eg create and update)